docs/src/guide/custom-project.md

```diff
@@ -0,0 +1,299 @@
+   `modm/src/modm/board/{name}` to your own project and modify them.
+
+2. In your `project.xml` remove the board config inheritance (`<extends>`) and
+   instead copy the pre-defined options into your own config.
+
+3. Check what modm modules you need to depend on and add them to your own
+   project (check for `module.depends(...)` in the BSPs `module.lb`).
+
+4. You may need to manually [add the pre-defined collector values to your project
+   configuration](../../reference/build-systems/#customization)
+   (check for `env.collect(...)` in the BSPs `module.lb`).
+
+
+## Custom Configuration
+
+For completely custom hardware, you start by definining the `modm:target` option
+and then adding the modules with their options as needed. For example, a basic
+configuration for the STM32F469NIH6 with startup, GPIO and clock modules looks
+like this:
+
+```xml
+<library>
+  <repositories>
+    <repository><path>../../ext/modm/repo.lb</path></repository>
+  </repositories>
+  <options>
+    <option name="modm:target">stm32f469nih6</option>
+  </options>
+  <modules>
+    <module>modm:platform:clock</module>
+    <module>modm:platform:core</module>
+    <module>modm:platform:gpio</module>
+    <module>modm:architecture:delay</module>
+    <module>modm:build:scons</module>
+    <module>modm:docs</module>
+  </modules>
+</library>
+```
+
+Remember to include the `modm:docs` module, since you need to generate your own
+API docs from your specific configuration:
+
+```sh
+(cd modm/docs && doxypress doxypress.json)
+# open modm/docs/html/index.html
+```
+
+A minimal `main.cpp` for this configuration can look like this. Since there is
+no BSP, you need to `#include <modm/platform.hpp>` and add the `modm::platform`
+namespace manually.
+
+```cpp
+#include <modm/platform.hpp>
+using namespace modm::platform;
+using namespace std::chrono_literals;
+
+int main()
+{
+    GpioA0::setOutput();
+    while (true)
+    {
+        GpioA0::toggle();
+        modm::delay(0.5s);
+    }
+}
+```
+
+We recommend to start your custom projects with a known-good configuration from
+one of our examples and then work your way into your specialization.
+
+!!! tip "Visualize your dependencies"
+    You can create a dependency graph with
+    `lbuild dependencies | dot -Tsvg -Grankdir=BT -o dependencies.svg`
+    to help you understand what code is pulled in when you generate your library.
+
+
+### Generate, Compile, Upload
+
+Once you have your `project.xml` set up, you can call `lbuild build`, which
+generates the target and configuration specific library from modm.
+This will create a few files and folders:
+
+```
+ $ cd app/project_name
+ $ ls
+main.cpp    project.xml
+ $ lbuild build
+ $ ls
+SConstruct      main.cpp        modm            project.xml     project.xml.log
+```
+
+For this project we included the `modm:build:scons` generator, so we can just
+call `scons build`, which will compile the entire source code and output the
+resource consumption:
+
+```
+ $ scons build
```

Comment:
Shouldn't we recommend using a parallel build with `-j`? Not everyone is familiar with SCons and knows it is supported.

Reply:
I think scons does that automatically?

Reply:
Not on my computer:
```
$ time scons build -j8
scons: done building targets.

real    0m10,984s
user    0m59,327s
sys     0m12,194s
```
But:
```
$ time scons build
scons: done building targets.

real    0m28,432s
user    0m23,997s
sys     0m4,486s
```

11s vs 28s

Reply:
Added it in a few places incl. examples/README.md, reference/build-systems and in guide/custom-project. It's already noted in the build system docs.